app/src/main/java/org/mozilla/focus/utils/Settings.kt

```diff
@@ -195,11 +195,18 @@ class Settings(
         preferences.edit()
             .putString(getPreferenceKey(R.string.pref_key_autoplay), prefKey)
             .apply()
-        context.components.sessionUseCases.reload.invoke(context.components.store.state.selectedTabId)
-        currentAutoplayOption = getValueByPrefKey(autoplayPrefKey = prefKey, context = context)
+        autoplayState = autoplayState.copy(
+            option = getValueByPrefKey(autoplayPrefKey = prefKey, context = context),
+            reloadSession = true
```

Comment:
This seems like misplaced responsibility.
If this is just "state" it should not dictate how the app will function - "reload session".
I understand it serve the purpose of just informing when the value has been "updated" so maybe it can be named as such?